### PR TITLE
🧪 Add explicit edge case tests for AudioPlaybackContext

### PR DESCRIPTION
**Pull request by is0692vs**
🎯 **What:** The testing gap in `AudioPlaybackContext` was addressed. Specifically, missing tests for `setPlaybackSource` edge cases were added, and `jest.config.js` was updated to collect coverage from the `contexts/` directory.

📊 **Coverage:** The following scenarios are now tested:
*   Stopping playback when the next source is `null`.
*   Not stopping playback when the next source has the same `articleUrl`.
*   Not stopping playback when the initial source is `null`.

✨ **Result:** Test coverage for `contexts/AudioPlaybackContext.tsx` is now complete and actively tracked in coverage reports, ensuring safer future refactoring.

---
*PR created automatically by Jules for task [5264161526948947119](https://jules.google.com/task/5264161526948947119) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

このPRは `AudioPlaybackContext` の `setPlaybackSource` に対する3つのエッジケーステスト（次のソースが `null`・同一URL・初期 `null`）を追加し、`jest.config.js` で `contexts/` ディレクトリをカバレッジ収集対象に追加します。

- **新規テスト3件**: `null` への切り替え時に `stop` が呼ばれること、同一 `articleUrl` では `stop` が呼ばれないこと、初期状態からの設定では `stop` が呼ばれないことをそれぞれ検証。実装の条件分岐と正確に対応しており、ロジックも正しい。
- **`jest.config.js`**: `contexts/**/*.{ts,tsx}` を `collectCoverageFrom` に追加。ただし `contexts/__tests__/**` の除外パターンが欠けており、テストファイル自体がカバレッジレポートに表示される可能性がある（詳細はインラインコメント参照）。
</details>

<details open><summary><h3>Confidence Score: 4/5</h3></summary>

テストロジックは正確で本番コードへの影響はなく、設定漏れはカバレッジレポートのノイズにとどまるためマージは安全。

`jest.config.js` でテストファイルがカバレッジ収集対象に含まれてしまう設定漏れがあるが、テストの動作・本番コードには影響せず、修正も1行で完結する軽微な問題。

packages/web-app-vercel/jest.config.js — `!contexts/**/__tests__/**` 除外パターンの追加が必要。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/web-app-vercel/contexts/__tests__/AudioPlaybackContext.test.tsx | `setPlaybackSource` の3つのエッジケース（null→null、同一URL、初期null）を正確にカバーする新規テストを追加。ロジックは実装と一致しており、`act()` によるステート更新のフラッシュも適切に処理されている。 |
| packages/web-app-vercel/jest.config.js | `contexts/**/*.{ts,tsx}` をカバレッジ収集対象に追加。ただし `contexts/__tests__/**` の除外パターンが抜けており、テストファイルがカバレッジレポートに混入する可能性がある。 |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A["setPlaybackSource(next)"] --> B{source is null?}
    B --Yes--> D[setSource next]
    B --No--> C{next is null or articleUrl changed?}
    C --No--> D
    C --Yes--> E[stopPlayback]
    E --> D

    subgraph NewTests["新規テスト"]
        T1["next=null → stop called"]
        T2["same articleUrl → stop not called"]
        T3["source=null → stop not called"]
    end
```

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 1 code review issue. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 1
packages/web-app-vercel/jest.config.js:29-30
`contexts/**/*.{ts,tsx}` パターンは `contexts/__tests__/AudioPlaybackContext.test.tsx` もカバレッジ収集対象に含めてしまいます。既存の `lib/` に対して `"!lib/**/__tests__/**"` という除外パターンが用意されているのと同様に、`contexts/__tests__` も除外する必要があります。このままではテストファイル自体がカバレッジレポートに「カバー済み」として表示され、ソースのカバレッジ計測が不正確になります。

```suggestion
    "contexts/**/*.{ts,tsx}",
    "!lib/**/__tests__/**",
    "!contexts/**/__tests__/**",
```

`````

</details>

<sub>Reviews (1): Last reviewed commit: ["test: add explicit edge case tests for A..."](https://github.com/hiroki-org/audicle/commit/d7a006565c3825bcb63d00e4c9d0c441881f279a) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=31480481)</sub>

> Greptile also left **1 inline comment** on this PR.

### CODE REVIEW

File: packages/web-app-vercel/contexts/__tests__/AudioPlaybackContext.test.tsx [hunk 1]
```diff
@@ -251,4 +251,147 @@ describe('AudioPlaybackContext', () => {
...
+    });
+
+    act(() => {
+      contextValue.setSource(null);
+    });
+
+    expect(mockStop).toHaveBeenCalledTimes(1);
+  });
+
+  it('should not stop playback when the next source has the same articleUrl', () => {
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/contexts/__tests__/AudioPlaybackContext.test.tsx:303]: 対応しました: setPlaybackSource の変更判定に voiceModel を含め、同一 articleUrl でも voiceModel が変わる場合は stop するようにしました。追加テストは DOM ボタン操作で voice A から voice B に切り替え、単一の waitFor 内で source 更新と stop 呼び出しを検証しています。

File: packages/web-app-vercel/jest.config.js [hunk 2]
```diff
@@ -26,7 +26,8 @@ const customJestConfig = {
   collectCoverageFrom: [
     "lib/**/*.{js,jsx,ts,tsx}",
     "components/DomainBadge.tsx",
-    "!lib/**/__tests__/**",
+    "contexts/**/*.{ts,tsx}",
+    "!**/__tests__/**",
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/jest.config.js:30]: 確認しました。coverage exclude の汎用化でこの点は対応済みです。

File: packages/web-app-vercel/jest.config.js [hunk 3]
```diff
@@ -26,6 +26,7 @@ const customJestConfig = {
   collectCoverageFrom: [
     "lib/**/*.{js,jsx,ts,tsx}",
     "components/DomainBadge.tsx",
+    "contexts/**/*.{ts,tsx}",
     "!lib/**/__tests__/**",
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/jest.config.js]: 対応しました: lib 固有の除外から !**/__tests__/** に変更し、ディレクトリを問わずテストファイルを coverage 対象外にしました。

File: packages/web-app-vercel/contexts/__tests__/AudioPlaybackContext.test.tsx [hunk 4]
```diff
@@ -251,4 +251,227 @@ describe('AudioPlaybackContext', () => {
 
     expect(mockStop).toHaveBeenCalledTimes(1);
   });
+
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply packages/web-app-vercel/contexts/__tests__/AudioPlaybackContext.test.tsx:254]: 確認しました。この指摘は既存テスト全体のスタイル統一に関する提案で、今回の差分では新規追加した voice model 変更テストを DOM 操作 + 単一 waitFor の形にしています。既存 contextValue パターンの全面置換はこのPRの範囲外として、別のテスト整理PRで扱うのが適切と判断します。
